### PR TITLE
Suggestions to tx_annotate_mt 

### PR DESCRIPTION
**Pull request by jkgoodrich**

### CODE REVIEW

File: gnomad/utils/transcript_annotation.py [hunk 1]
```diff
@@ -177,93 +181,67 @@ def tissue_expression_ht_to_array(
...
-                ht.vep_processed_csqs.most_severe_consequence
-            )
-        )
-
-    # TODO: Need to modify process consequences to ignore splice variants,
-    #  because these can occur on intronic regions?
-    splicing = hl.literal(
-        ["splice_acceptor_variant", "splice_donor_variant", "splice_region_variant"]
-    )
+    splice_csqs = [
```
Comment (jkgoodrich) [inline gnomad/utils/transcript_annotation.py:207]: Maybe we should add this as a global utils.vep? similar to the others like `CSQ_CODING_HIGH_IMPACT`, `CSQ_CODING_MEDIUM_IMPACT`, `CSQ_CODING_LOW_IMPACT`, ...

**Review by jkgoodrich — COMMENTED** [review]
A few more comments/ questions

File: gnomad/utils/transcript_annotation.py [hunk 2]
```diff
@@ -177,93 +181,67 @@ def tissue_expression_ht_to_array(
...
-        For example, if you want to annotate each variant with the worst
-        consequence in each gene it falls on and the transcript expression,
-        you would use "worst_csq_by_gene". Default is "transcript_consequences".
-    :return: Table of variants that fall on transcripts of interest.
+    :param filter_to_protein_coding: Whether to filter to protein coding transcripts.
+        Default is True.
+    :param vep_root: Name used for root VEP annotation. Default is 'vep'.
+    :return: Table of transcript expression information prepared for annotation.
     """
     # TODO: Filter to only CDS regions?
```
[the remark this answers is not in the document]
Comment (KoalaQin) [reply gnomad/utils/transcript_annotation.py:205]: No, it seems that they filtered to 'CDS' from the Gencode GTF to make a bed file, Beryl used a bed file directly, but I don't get the same number of 'CDS' lines based on Konrad's code to import_gtf, need to dig into it.

File: gnomad/utils/transcript_annotation.py [hunk 3]
```diff
@@ -177,93 +181,67 @@ def tissue_expression_ht_to_array(
...
-        ht = ht.filter(ht.freq[0].homozygote_count > 0)
-
-    if filter_to_protein_coding:
-        ht = ht.filter(ht.vep_processed_csqs.biotype == "protein_coding")
-
-    return ht
+        if filter_to_csqs is not None:
+            filter_to_csqs = [csq for csq in filter_to_csqs if csq not in splice_csqs]
+        else:
+            # TODO: Need to modify process consequences to ignore splice variants,
```
[the remark this answers is not in the document]
Comment (KoalaQin) [reply gnomad/utils/transcript_annotation.py:217]: I discussed with Konrad about keeping 'OS' variants, we think current csq_order to prioritze 'OS' over 'LC' is better than deleting them, but I keep that as an option in the pull out worst after tx annotation.

File: gnomad/utils/transcript_annotation.py [hunk 4]
```diff
@@ -177,93 +181,67 @@ def tissue_expression_ht_to_array(
 
 def preprocess_variants_for_tx(
```
[the remark this answers is not in the document]
Comment (KoalaQin) [reply gnomad/utils/transcript_annotation.py:182]: Great idea! I didn't know that function exists.

File: gnomad/utils/transcript_annotation.py [hunk 5]
```diff
@@ -177,93 +181,67 @@ def tissue_expression_ht_to_array(
...
+    filter_to_csqs: Optional[List[str]] = None,
     ignore_splicing: bool = True,
-    filter_to_homs: bool = False,
     filter_to_protein_coding: bool = True,
-    vep_annotation: str = "transcript_consequences",
+    vep_root: str = "vep",
 ) -> hl.Table:
     """
-    Filter variants to those that fall on transcripts of interest.
+    Prepare a Table of transcript expression information for annotation.
```
Comment (KoalaQin) [inline gnomad/utils/transcript_annotation.py:192]: Why is it a table of transcript expression information? I think this is a variant table without any expression information.

**Pull request merged**